shinysdr/plugins/wspr/telemetry.py

```diff
@@ -0,0 +1,134 @@
+"""ShinySDR telemetry for WSPR"""
+
+from __future__ import division, absolute_import
+
+import time
+from collections import namedtuple
+
+from zope.interface import implementer, Interface
+
+from shinysdr.telemetry import (
+    ITelemetryMessage, ITelemetryObject, Track, TelemetryItem)
+from shinysdr.types import TimestampT
+from shinysdr.values import ExportedState, exported_value
+
+
+MINUTES = 60
+
+
+@implementer(ITelemetryMessage)
+class WSPRSpot(namedtuple('WSPRSpot', [
+    'time',
+    'snr',
+    'dt',
+    'frequency',
+    'drift',
+    'call',
+    'grid',
+    'txpower',
+])):
+    def get_object_id(self):
+        return 'wsprspot_%s_%s' % (self.call, self.grid)
+
+    def get_object_constructor(self):
+        return WSPRStation
+
+
+class IWSPRStation(Interface):
+    pass
+
+
+@implementer(ITelemetryObject, IWSPRStation)
+class WSPRStation(ExportedState):
+    __snr = None
+    __frequency = None
+    __call = None
+    __grid = None
+    __txpower = None
+
+    def __init__(self, object_id):
+        self.__object_id = object_id
+
+    def receive(self, message):
+        self.__last_heard = message.time
+        self.__snr = message.snr
+        self.__frequency = message.frequency
+        self.__call = message.call
+        self.__grid = message.grid
+        self.__txpower = message.txpower
```

Comment:
I fixed: since you are declaring changes='explicit', you must call self.state_changed() here.

Several of your getters errored out under test because of fields being None; updated them to handle that.